### PR TITLE
Use new Rails Engine routes if facilityLocatorRailsEngine feature flag is true

### LINKED ISSUE #22701: Front end route changes for Rails engine
## Issue Description
Need to change routes to complete Rails engine work

---

## Tasks
- [x] Add flipper to control routes
- [x] Change route for each API call to new route `https://api.va.gov/facilities_api/v1/ccp` 
- [x] **For ccp routes only**, the service type parameter will go away (i.e. provider, urgent care) and will now be part of route. i.e.:
  - `https://api.va.gov/facilities_api/v1/ccp/urgent_care`
  - `https://api.va.gov/facilities_api/v1/ccp/provider`
  - `https://api.va.gov/facilities_api/v1/ccp/pharmacy`

## Acceptance Criteria
- [x] Calls use new routes
---

### PR DESCRIPTION
**Pull request by mpelzsherman**
Resolves https://github.com/department-of-veterans-affairs/va.gov-team/issues/22701

### CODE REVIEW

File: src/platform/site-wide/feature-toggles/reducers/index.js [hunk 1]
```diff
@@ -14,12 +14,16 @@ export const FeatureToggleReducer = (state = INITIAL_STATE, action) => {
         loading: true,
       };
     case FETCH_TOGGLE_VALUES_SUCCEEDED:
+      window.VetsGov = window.VetsGov || {};
```
Comment (brangi) [inline src/platform/site-wide/feature-toggles/reducers/index.js:17]: Does this mean, we will have 2 ways of getting feature flags? the redux store way and through the window object?

**Review by timwright12 — DISMISSED** [review]
In the process of discussing the safety and necessity of including feature toggles in the window object.

File: src/platform/site-wide/feature-toggles/reducers/index.js (hunk 1 above)
Comment (mpelzsherman) [reply src/platform/site-wide/feature-toggles/reducers/index.js:17]: Yes, at least initially. Eventually we could make the window way the "official" way.

**Comment by mpelzsherman** [comment]
@timwright12 I will make an alternate PR to demonstrate the alternative (passing the prop through everywhere we make a query). Also, maybe I can come up with some way of connecting our config object to redux. But I think making the feature flags available on the window object would be super simple and convenient. Thanks for the consideration, and I'll be discussing with @mmiddaugh how to proceed here.

**Comment by mpelzsherman** [comment]
See https://github.com/department-of-veterans-affairs/vets-website/pull/17386 for alternate approach (passing prop through to the API config from the container components).

**Comment by alexandec** [comment]
@mpelzsherman Could you try importing the store and getting the value from it? See [this post](https://daveceddia.com/access-redux-store-outside-react/) for more information. That way you'll avoid having a duplicate set of data and you won't need to put anything in the window object.

**Comment by mpelzsherman** [comment]
I saw that post @alexandec but I don't know how to apply it to our codebase. If you have time to pair with me I'd really appreciate it. Thanks! 😄

**Comment by alexandec** [comment]
@mpelzsherman I had a chance to look at this a little more. You can return the store from the `startApp()` function (currently it returns undefined) https://github.com/department-of-veterans-affairs/vets-website/blob/7c1289a45ceaaba643aa45a0f43aafcdb4e2433d/src/platform/startup/index.js#L57

Then in the facility locator entry file you can export the store with `export const store = startApp({ ... })` https://github.com/department-of-veterans-affairs/vets-website/blob/7c1289a45ceaaba643aa45a0f43aafcdb4e2433d/src/js/facility-locator/facility-locator-entry.jsx#L10 

That will make the store available for import wherever you need it in the Facility Locator app. You can get the store state with `store.getState()` and you can listen for changes with `store.subscribe()`.

@cvalarida @U-DON does this approach look good to you?

**Comment by cvalarida** [comment]
The approach of importing the redux store directly seems like the best approach. It'd reduce the surface area for weird bugs by keeping only a single source of truth for feature toggle states.

That said, I'd be a little nervous to import anything from the entry file myself. It'll create a cyclical dependency. Instead, what I'd recommend is getting `startApp` to accept a pre-made Redux store _or_ create its own like it's doing now. Then in your application, you'd create that Redux store with everything you need (just copy what `startApp` is doing now) in a new module, and import the store from that module. That'd avoid cyclical module dependencies.

**Comment by U-DON** [comment]
I’m pretty rusty with Redux at this point, but it sounds like the main point of exposing the store is to change the API endpoint the actions are using. Thunk action creators of the form `(dispatch, getState) => { ... }`, though the actions you’re using now only make use of `dispatch`. `LocatorApi`’s methods are only invoked from those actions. You may be able to refactor `LocatorApi` to have a constructor that accepts some legacy flag, and in those actions, you could do:

```js
const state = getState();
const shouldUseLegacyApi = state.featureToggles.yourToggleName;
LocatorApi(shouldUseLegacyApi).fetchVAFacility(id);
```

**Comment by mpelzsherman** [comment]
Thanks for all of the ideas, folks! I decided to give @alexandec 's suggestion a try and it seems to be working!

It's a bit tricky to test this locally because of issues with the dev api but I think this is the way to go. I don't see any issues with circular dependencies. <shrug?>

I don't see an immediate need to refactor the `LocatorApi` per your suggestion, @U-DON, but that would be a nice cleanup down the road.

File: src/platform/site-wide/feature-toggles/reducers/index.js (hunk 1 above)
Comment (mpelzsherman) [reply src/platform/site-wide/feature-toggles/reducers/index.js]: Decided to go with grabbing the redux store from `startApp` instead.

**Review by brangi — APPROVED** [review]

**Comment by alexandec** [comment]
One other note, there may be the potential for a race condition since if the feature flag request is slow, this code may run before the request completes. Sometimes that can occur if the use reloads the page for example. Not sure if it's an issue in this particular case but wanted to mention it

**Comment by mpelzsherman** [comment]
The race condition issue exists for almost every usage of the feature flag. It's not a huge concern for this feature because worst-case-scenario, the feature flag will revert to the old behavior. But for other cases, the race condition is definitely an issue, which is why I created [this ticket](https://github.com/department-of-veterans-affairs/va.gov-team/issues/25105).

**Review by jhonny218 — APPROVED** [review]

**Pull request merged**